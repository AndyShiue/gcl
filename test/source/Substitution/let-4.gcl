var y : Int 
let Q x = y
let P x y = x = y

y := y + 1

{ P 3 (Q 1) }
