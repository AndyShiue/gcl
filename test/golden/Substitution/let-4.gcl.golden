P [ y + 1 / y ] 3 (Q [ y + 1 / y ] 1)
  P [ y + 1 / y ] 3 (Q [ y + 1 / y ] 1) ===> 3 = Q [ y + 1 / y ] 1
    Q [ y + 1 / y ] 1
      Q [ y + 1 / y ] 1 ===> y + 1
        
  Q [ y + 1 / y ] 1 ===> y + 1
    